### PR TITLE
Create new button component with unified styles

### PR DESCRIPTION
**Pull request by pcardune**
Here is what they look like:
screen shot 2016-06-30 at 1 00 46 pm

### CODE REVIEW

**Comment by pcardune** [comment]
cc @poorvasingal @breville @tanyaparker

File: apps/src/templates/Button.jsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+      right: 10 - ARROW_WIDTH,
+      borderLeftColor: color,
+    }),
+  },
+};
+
+style.withArrow = {
+  base: {
+    position: 'relative',
+    top: (ARROW_HEIGHT - style.base.borderWidth - 10 * 2 - 44)/2,
```
Comment (islemaster) [inline apps/src/templates/Button.jsx]: Magic numbers?

File: apps/src/templates/Button.jsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  children: React.PropTypes.node,
+  size: React.PropTypes.oneOf(['normal', 'large']),
+};
+
+const ArrowButton = Radium(function ArrowButton(props) {
+  const config = BUTTON_TYPES[props.type];
+  if (process.env.NODE_ENV === 'development') {
+    if (!props.size === 'large') {
+      throw new Error(
+        `Arrows can only be used with large buttons. This button is "${props.size}" can't use arrows.`
```
Comment (islemaster) [inline apps/src/templates/Button.jsx]: Fragment? Maybe `This "${props.size}" button can't use arrows.`

File: apps/src/templates/Button.jsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+  const config = BUTTON_TYPES[props.type];
+  let styleArray = [style.base, config.style, sizeStyle];
+  return (
+    <button {...props} style={[styleArray, props.style]}>
+      {props.children}
+    </button>
+  );
+});
+BaseButton.propTypes = {
+  type: React.PropTypes.oneOf(Object.keys(BUTTON_TYPES)),
```
Comment (islemaster) [inline apps/src/templates/Button.jsx:118]: Slick.

File: apps/src/templates/Button.jsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        style.withArrow[props.arrow],
+      ]}/>
+    </div>
+  );
+});
+ArrowButton.propTypes = Object.assign({}, BaseButton.propTypes, {
+  arrow: React.PropTypes.oneOf(['left', 'right']).isRequired,
+});
+
+const Button = Radium(function Button(props) {
```
Comment (islemaster) [inline apps/src/templates/Button.jsx:146]: Why are you using a named function here, that you immediately clobber with the Radium-ized version?

**Comment by islemaster** [comment]
Pretty cool.  A few code style nits, otherwise LGTM.

**Comment by breville** [comment]
If we're up for modifying the visuals, could we experiment with smaller heads on the arrows?  And also vertically centering the text in them?  

And try adding some more left/right padding on the large buttons?

Also curious whether the text should be a little smaller on the large buttons & arrow buttons?

File: apps/src/templates/Button.jsx (hunk 4 above)
Comment (pcardune) [inline apps/src/templates/Button.jsx:146]: By using a named function, the JSX transpiler will give the component a displayName, which in turn shows up in debugging.

**Comment by poorvasingal** [comment]
@pcardune - nice work :D  Not sure if you've seen the comments in the buttons semantics doc but we're thinking of changing the color of the action button to our cyan.  The purple color doesn't work well outside of Code Studio.  Hadi also wants us to reconsider the colors for the default button + the cancel button.   Would that be an easy change to make?

**Comment by pcardune** [comment]
@poorvasingal Making the change to the colors for this particular component is easy. But if we want to change the colors everywhere on the site where those colors are used, that will require some careful testing/review to make sure there weren't any places that copy/pasted the color code rather than using the shared scss variable.

So I'd rather have this diff go through without modifying the colors, and then do a site wide color change in a separate diff/pull request/release.

**Comment by pcardune** [comment]
By the way @breville, my response to your question is the same as for @poorvasingal's question. I'm happy to experiment with changing the way this component (and all the buttons on the site) look, but I'd rather do that in a separate "style only" diff. This component matches what is on the site today (the good and the bad) and I don't want to put too many changes in one diff/release.

File: apps/src/templates/Button.jsx (hunk 1 above)
Comment (pcardune) [inline apps/src/templates/Button.jsx]: yeah, they are from common.scss. Will stick them in consts.

**Comment by pcardune** [comment]
Ok, just rebased. Will merge once tests pass (again).

**Pull request merged**